src/Rendering/depthPeelingRenderer.ts

```diff
@@ -0,0 +1,228 @@
+            } as RenderTargetCreationOptions,
+        ];
+
+        for (let i = 0; i < 2; i++) {
+            const depthTexture = this._engine._createInternalTexture(size, optionsArray[0]);
+            const frontColorTexture = this._engine._createInternalTexture(size, optionsArray[1]);
+            const backColorTexture = this._engine._createInternalTexture(size, optionsArray[1]);
+
+            // TODO : lower level (move in engine.multiRender)
+            this._engine.bindTextureFramebuffer(this._depthMrts[i].getInternalTexture()!._framebuffer as WebGLFramebuffer, depthTexture);
+            this._engine.bindTextureFramebuffer(this._depthMrts[i].getInternalTexture()!._framebuffer as WebGLFramebuffer, frontColorTexture, 1);
+            this._engine.bindTextureFramebuffer(this._depthMrts[i].getInternalTexture()!._framebuffer as WebGLFramebuffer, backColorTexture, 2);
+
+            this._engine.bindTextureFramebuffer(this._colorMrts[i].getInternalTexture()!._framebuffer as WebGLFramebuffer, frontColorTexture);
+            this._engine.bindTextureFramebuffer(this._colorMrts[i].getInternalTexture()!._framebuffer as WebGLFramebuffer, backColorTexture, 1);
+        }
+
+        const blendBackTexture = this._engine._createInternalTexture(size, optionsArray[1]);
+        this._engine.bindTextureFramebuffer(this._blendBackMrt.getInternalTexture()!._framebuffer as WebGLFramebuffer, blendBackTexture);
+    }
+
+    private _createEffects() {
+        this._blendBackEffect = this._scene.getEngine().createEffect(
+            {
+                vertex: "postprocess",
+                fragment: "oitBackBlend",
+            },
+            {
+                attributes: [],
+                samplers: [],
+                uniformsNames: [],
+                uniformBuffersNames: [],
+                defines: "",
+                fallbacks: null,
+                onCompiled: null,
+                onError: null,
+            },
+            this._engine
+        );
+
+        this._finalEffect = new Effect(
+            {
+                vertex: "postprocess",
+                fragment: "oitFinal",
+            },
+            {
+                attributes: [],
+                samplers: [],
+                uniformsNames: [],
+                uniformBuffersNames: [],
+                defines: "",
+                fallbacks: null,
+                onCompiled: null,
+                onError: null,
+            },
+            this._engine
+        );
+    }
+
+    private _updateSize() {
+        // TODO
+    }
+
+    public render(transparentSubMeshes: SmartArray<SubMesh>) {
+        if (!this._blendBackEffect.isReady() || !this._finalEffect.isReady()) {
+            return;
+        }
+
+        const DEPTH_CLEAR_VALUE = -99999.0;
+        const MIN_DEPTH = 0;
+        const MAX_DEPTH = 1;
+
+        // Clears
+        this._engine._bindUnboundFramebuffer(this._blendBackMrt.getInternalTexture()!._framebuffer as WebGLFramebuffer);

```

Comment:
We should not get WebGL access in here (cc @Popov72 to confirm)

Reply:
yep this is really an early preview for discussion, all webgl calls will be added in thinEngine. In that case we will have a `MultiRenderTarget.setTexture(index: number)` to perform the framebuffer layout